bin/dotc

```diff
@@ -57,11 +57,21 @@ function checkjar {
     else
       echo "The required jar file was built successfully."
     fi
+  else
+    NEW_FILES="$(find "$DOTTY_ROOT/$3" -iname "*.scala" -newer "$1")"
+    if [ ! -z "$NEW_FILES" ]; 
+    then
+      echo "new files detected. rebuilding"
+      cd $DOTTY_ROOT
+      sbt $2
+      touch "$1"
```

Comment:
Sometimes sbt is smart enough to see that it does not need to rebuild the project, though timestamps changed. Eg if you make a change and revert it.